gdb breaks when -bbigtoc linker flag is necessary.
C++ exception handling breaks when libstdc++ and/or libgcc_s
is built with -mminimal-toc.

http://bugs.gentoo.org/show_bug.cgi?id=266483
So reduce TOC size, but avoid -mminimal-toc.

--- gcc-4.2.4/gcc/config/rs6000/rs6000.c.orig	2011-06-20 16:01:28 +0200
+++ gcc-4.2.4/gcc/config/rs6000/rs6000.c	2011-06-20 16:04:53 +0200
@@ -1229,6 +1229,10 @@
 		     | MASK_DLMZB)
   };
 
+  /* gdb breaks when -bbigtoc linker flag is necessary */
+  TARGET_NO_FP_IN_TOC = 1;
+  TARGET_NO_SUM_IN_TOC = 1;
+
   rs6000_init_hard_regno_mode_ok ();
 
   set_masks = POWER_MASKS | POWERPC_MASKS | MASK_SOFT_FLOAT;
